### PR TITLE
[MRG] use travis build with minimal deps

### PR DESCRIPTION
**Pull request by agramfort**

### CODE REVIEW

**Comment by agramfort** [comment]
We don’t use Travis in this repo ???

**Comment by sappelhoff** [comment]
we do :-) see: https://github.com/mne-tools/mne-bids/blob/master/.travis.yml

you even made a PR some weeks ago to add MacOS to the build

File: .travis.yml [hunk 1]
```diff
@@ -55,14 +64,17 @@ install:
...
-      pip install --no-deps -e .
-      cd ../
+    - if [ "${DEPS}" != "minimal" ]; then
+          conda env create -f $CONDA_ENV --name=testenv
+          source activate testenv
+          pip uninstall --yes --quiet mne  # uninstall the pip version of mne to install dev version below
+          git clone --depth 1 https://github.com/mne-tools/mne-python.git -b $MNE_VERSION
+          cd mne-python
+          pip install --no-deps -e .
+          cd ../
```
Comment (sappelhoff) [inline .travis.yml]: I would pull these lines out of the if/else logic and put it below. Even for minimal dependencies, we might want to set the MNE version to be used with our `$MNE_VERSION` env variable

**Comment by agramfort** [comment]
>
> I would pull these lines out of the if/else logic and put it below. Even
> for minimal dependencies, we might want to set the MNE version to be used
> with our $MNE_VERSION env variable
>
our maint branches are not always up to date with pypi

i'd like to make sure it works with pypi version.

**Comment by agramfort** [comment]
@jasmainak @sappelhoff error is legit. Pb is that data files are too big to be pushed to pypi.

2 options:
- require people to use a dev version of mne
- using the mne testing data

thoughts?

**Comment by jasmainak** [comment]
@agramfort you need those files only for tests. My recommendation would be to use the `dev` or `maint/versionxxx` branch by the developers. Users don't need to run tests.

**Comment by agramfort** [comment]
green. Feel free to MRG @jasmainak or @sappelhoff

**Comment by jasmainak** [comment]
LGTM merged. Thanks @agramfort !

**Comment by sappelhoff** [comment]
Thanks a lot @agramfort!

**Pull request merged**